src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -1125,19 +1115,7 @@ private void synthesizeValueMethod(JCMethodDecl methodDecl) {
                     indyType,
                     List.nil().toArray(new LoadableConstant[0]));
 
-
             switch (methodDecl.name.toString()) {
```

Comment:
This switch and the case "toString" should be deleted since it is completely unnecessary. Earlier switch would ensure that the only way control can flow into this point is when the method concerned is toString() so the check is redundant